Global.h

```diff
@@ -279,442 +279,397 @@ void inline handleException()
-	typename Container::const_iterator find(const Container & c, const Item &i)
-	{
-		return std::find(c.begin(),c.end(),i);
-	}
+//returns iterator to the given element if present in container, end() if not
+template<typename Container, typename Item> typename Container::iterator find(Container & c, const Item & i)
+{
+	return std::find(c.begin(), c.end(), i);
+}
 
-	//removes element i from container c, returns false if c does not contain i
-	template <typename Container, typename Item>
-	typename Container::size_type operator-=(Container &c, const Item &i)
-	{
-		typename Container::iterator itr = find(c,i);
-		if(itr == c.end())
-			return false;
-		c.erase(itr);
-		return true;
-	}
+//returns const iterator to the given element if present in container, end() if not
+template<typename Container, typename Item> typename Container::const_iterator find(const Container & c, const Item & i)
+{
+	return std::find(c.begin(), c.end(), i);
+}
 
-	//assigns greater of (a, b) to a and returns maximum of (a, b)
-	template <typename t1, typename t2>
-	t1 &amax(t1 &a, const t2 &b)
-	{
-		if(a >= b)
-			return a;
-		else
-		{
-			a = b;
-			return a;
-		}
-	}
+//removes element i from container c, returns false if c does not contain i
+template<typename Container, typename Item> typename Container::size_type operator-=(Container & c, const Item & i)
+{
+	typename Container::iterator itr = find(c, i);
+	if(itr == c.end())
+		return false;
+	c.erase(itr);
+	return true;
+}
 
-	//assigns smaller of (a, b) to a and returns minimum of (a, b)
-	template <typename t1, typename t2>
-	t1 &amin(t1 &a, const t2 &b)
+//assigns greater of (a, b) to a and returns maximum of (a, b)
+template<typename t1, typename t2> t1 & amax(t1 & a, const t2 & b)
+{
+	if(a >= b)
+		return a;
+	else
 	{
-		if(a <= b)
-			return a;
-		else
-		{
-			a = b;
-			return a;
-		}
+		a = b;
+		return a;
 	}
+}
 
-	//makes a to fit the range <b, c>
-	template <typename t1, typename t2, typename t3>
-	t1 &abetween(t1 &a, const t2 &b, const t3 &c)
+//assigns smaller of (a, b) to a and returns minimum of (a, b)
+template<typename t1, typename t2> t1 & amin(t1 & a, const t2 & b)
+{
+	if(a <= b)
+		return a;
+	else
 	{
-		amax(a,b);
-		amin(a,c);
+		a = b;
 		return a;
 	}
+}
 
-	//checks if a is between b and c
-	template <typename t1, typename t2, typename t3>
-	bool isbetween(const t1 &value, const t2 &min, const t3 &max)
-	{
-		return value > min && value < max;
-	}
+//makes a to fit the range <b, c>
+template<typename t1, typename t2, typename t3>	t1 & abetween(t1 & a, const t2 & b, const t3 & c)
+{
+	amax(a, b);
+	amin(a, c);
+	return a;
+}
 
-	//checks if a is within b and c
-	template <typename t1, typename t2, typename t3>
-	bool iswithin(const t1 &value, const t2 &min, const t3 &max)
-	{
-		return value >= min && value <= max;
-	}
+//checks if a is between b and c
+template<typename t1, typename t2, typename t3>	bool isbetween(const t1 & value, const t2 & min, const t3 & max)

```

Comment:
One more example of template "one lining" - this is not helpful at all imo. Moreover we go for two-liners with constructor's initializer lists etc so why opposite approach.